sote/game/scenes/game.lua

```diff
@@ -812,16 +814,278 @@ function gam.draw()
+		if gam.texture_atlas == nil then
+			gam.texture_atlas = love.graphics.newImage("textures/atlas.png")
+			gam.texture_atlas:setFilter("nearest", "nearest")
+		end
+		gam.planet_shader:send('texture_atlas', gam.texture_atlas)
+	end
+
+
+	if gam.planet_shader:hasUniform("texture_sprawl_frequency") then
+		if gam.DATA_TEXTURES_CACHE["texture_sprawl_frequency"] == nil then
+			local frequency_image_data = love.image.newImageData(256, 256, "rgba8")
+			local pointer_province_color = require("ffi").cast("uint8_t*", frequency_image_data:getFFIPointer())
+			local id = 0
+
+			for _, province in ipairs(WORLD.ordered_provinces_list) do
+
+				pointer_province_color[id * 4 + 0] = 255 * math.min(1, province:local_population() / 200)
+				pointer_province_color[id * 4 + 1] = 255 * 0
+				pointer_province_color[id * 4 + 2] = 255 * 0
+				pointer_province_color[id * 4 + 3] = 255 * 0
+
+				id = id + 1
+			end
+
+			local frequency_image = love.graphics.newImage(frequency_image_data)
+			frequency_image:setFilter("nearest", "nearest")
+
+			gam.DATA_TEXTURES_CACHE["texture_sprawl_frequency"] = frequency_image
+		end
+
+		gam.planet_shader:send('texture_sprawl_frequency', gam.DATA_TEXTURES_CACHE["texture_sprawl_frequency"])
+	end
+
+	if gam.planet_shader:hasUniform("show_terrain") then
+		if gam.map_mode_data[gam.map_mode][7] == nil then
+			gam.planet_shader:send("show_terrain", 0)
+		elseif gam.map_mode_data[gam.map_mode][7] == mmut.MAP_MODE_TERRAIN_TEXTURE_INTERACTION.SHOW_TERRAIN then
+			gam.planet_shader:send("show_terrain", 1)
+		elseif gam.map_mode_data[gam.map_mode][7] == mmut.MAP_MODE_TERRAIN_TEXTURE_INTERACTION.HIDE_TERRAIN then
+			gam.planet_shader:send("show_terrain", 0)
+		end
+	end
+
+	if gam.planet_shader:hasUniform("texture_index_cubemap") then
+		if gam.DATA_TEXTURES_CACHE["texture_index_cubemap"] == nil then
+
+			local image = love.image.newImageData(WORLD.world_size * 3, WORLD.world_size * 3, "rgba8")
+
+			for _, current_tile in pairs(WORLD.tiles) do
+				local temp_i, temp_j = gam.tile_id_to_color_coords(current_tile)
+
+				local sprawl_heat = 0
+				local local_province = current_tile:province()
+
+				if local_province ~= nil then
+					local center = local_province.center
+					local distance = current_tile:distance_to(center)
+					sprawl_heat = math.min(1, 1 / distance)
+				end
+
+
+				if (current_tile.biome ~= nil) then
+
+					local is_peak = false
+					if current_tile.biome.name == "barren-mountainside" or
+						current_tile.biome.name == "rugged-mountainside" or
+						current_tile.biome.name == "mountainside-scrub" then
+						is_peak = true
+						for neigh in current_tile:iter_neighbors() do
+							if current_tile.elevation < neigh.elevation then
+								is_peak = false
+							end
+						end
+					end
+
+					local image_index_scaler = 1 / 64
+
+					local texture_index = 0
+
+					local is_sea = 0
+
+					if current_tile.biome.name == "tundra" then
+						texture_index = 29
```

Comment:
Are these indices an enum? If so, it should be one, it's difficult to reason about this way, imho